packages/contracts/test/registry.ts

```diff
@@ -1,46 +1,58 @@
-import {expect} from 'chai';
-import {ethers} from 'hardhat';
-import {customError} from './test-utils/custom-error-helper';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { customError } from './test-utils/custom-error-helper';
 
-const EVENTS = {
-  NewDAORegistered: 'NewDAORegistered',
-};
+import { DAO, Registry } from '../typechain';
+
+const REGISTER_DAO_ROLE = ethers.utils.id('REGISTER_DAO_ROLE')
 
 describe('Registry', function () {
-  let registry: any;
+  let registry: Registry;
+  let dao: DAO;
   let ownerAddress: string;
 
-  before(async () => {
-    const signers = await ethers.getSigners();
+  beforeEach(async function () {
+    const signers = await ethers.getSigners()
     ownerAddress = await signers[0].getAddress();
-  });
 
-  beforeEach(async function () {
+    const DAO = await ethers.getContractFactory('DAO');
+    dao = await DAO.deploy();
+    await dao.initialize('0x', ownerAddress, ethers.constants.AddressZero);
+
     const Registry = await ethers.getContractFactory('Registry');
     registry = await Registry.deploy();
+    await registry.initialize(dao.address);
+
+    await dao.grant(registry.address, ownerAddress, REGISTER_DAO_ROLE);
```

Comment:
I'm not sure I follow. Why are we asking a DAO to grant permissions to the new Registry?

Reply:
In this test, `dao` is the associated DAO managing the permissions (which in our use-case would be the AdminDAO). This DAO manages the permissions the through its ACL. Here we give the `ownerAddress` (here it is a simple wallet) the permission to execute 

https://github.com/aragon/core/blob/6f7d0c82ee39cd0a30007b239215a4e8426904a9/packages/contracts/contracts/registry/Registry.sol#L42-L46

at `registry.address` through the `REGISTER_DAO_ROLE`.

Reply:
Given the many related questions about this, I would strongly suggest we rename these parameters to be something like `adminDao` and/or `targetDao`. Otherwise it gets confusing.

Reply:
In the tests, the smart contracts, or both?

Reply:
Everywhere :)
Names should be meaningful/unambiguous everywhere. In this case, the variable names (but more in general, function and contract names too).

Reply:
In general I'd say, let's make sure that the new code already uses 100% meaningful names, but let's still not rename existing things that we are not explicitly changing.... at least until the future proof registries/factories are in place